admin-perspective/olm/install-forcenamespace/install-forcenamespace.md

```diff
@@ -0,0 +1,82 @@
+
+This design is not intended to convey every combination of controls that is possible via operator InstallModes, that information is available via [this spreadsheet](https://docs.google.com/spreadsheets/d/19J7IEjORg4-U6nQEwQkxPwe3SJazgpobN2HFKE2rFcc/edit?folder=1y5NhdQNm8b-NeX1I0uN6PHQzhp_NJPh0#gid=0). Controls in this design will be enabled/disabled/hidden as described in the spreadsheet and as conveyed in the screens below and the existing UI’s paradigms. This design is intended to show the location of controls and the interactions between them.
+
+This design also captures the ability to enable cluster monitoring on namespaces that are installed to and the operator requests it. This setting applies to only certain whitelisted namespaces, and the user should be warned of the implications of doing so. Operators can request monitoring be enabled on their install namespace but we will also present this option.
+
+## Installing with no operator-defined namespace (existing behavior)
+
+![Operator namespace not defined all](img/1-1-noDefinedNS-all.png)
+- Installing an operator without a operator-defined namespace now conveys that by default the operator would be installed on the **openshift-operators** namespace with the option to **Edit**.
+- Namespaces available to install to would not yet have an OperatorGroup object present configured to watch all namespaces.
+
+![Operator namespace not defined single](img/1-2-noDefinedNS-single.png)
+-A specific namespace may be chosen for the operator to be available to (and installed to.)
+
+![Operator namespace not defined single](img/1-3-noDefinedNS-single.png)
+-The install namespace will match the specific namespace the operator is available to, until the install namespace is manually changed.
+
+![Operator namespace not defined single](img/1-4-noDefinedNS-single.png)
+- The user can **Edit** to modify the namespace the operator is installed to.
+
+![Operator namespace not defined single](img/1-5-noDefinedNS-single.png)
+- The user manually sets an alternative namespace to install to.
+
+![Operator namespace not defined single](img/1-6-noDefinedNS-single.png)
+- Changing the availability namespace no longer has any affect on the install namespace, as it has been manually changed.
+
+## Installing with an operator-defined namespace
+
+![Operator namespace defined all](img/2-1-definedNS-all.png)
+- When an operator-defined installation namespace exists, that namespace will be the default install namespace for the operator.
+- The operator-defined namespace can be used for **all namespace** availability, or **specific namespace** availability, as the operator's InstallModes allow.
+- If the user **Edit** the install namespace, the namespaces available for selection besides the operator-defined namespace would be namespaces that not yet have an OperatorGroup object present configured to watch all namespaces.
```

Comment:
@dmesser @jhadvig This would be the same set as the comment above (1-1), just defaulting to the definedNamespace.

Reply:
If the user selects **all namespace** availability then the dropdown for install namespace must be filtered and exclude namespaces that have OperatorGroups present **not** configured to watch all namespaces. That's because there can only be one `OperatorGroup` per namespace.

Likewise, if the user selects **single namespace** availability, then the dropdown for install namespace must be filtered and exclude namespace that have an `OperatorGroup` present that is configured to watch all namespaces.

Reply:
@dmesser 
> If the user selects all namespace availability then the dropdown for install namespace must be filtered and exclude namespaces that have OperatorGroups present not configured to watch all namespaces.

otherwords, dropdown should only contain namespaces that all of the OperatorGroups are watching. right ? 
In true, does the Operator Install Namespace also contain only those namespaces ? Or all available namespaces ?

> if the user selects single namespace availability, then the dropdown for install namespace must be filtered and exclude namespace that have an OperatorGroup present that is configured to watch all namespaces.

otherwords, dropdown should only contain namespaces that all of the OperatorGroups are watching, **except** those namespaces which OperatorGroup is watching all the namespaces(in this case `openshift-operators`)? right?

Reply:
Not quite because the UI would also create `OperatorGroup` objects if none exist in the namespace yet. That's why mantra is:

- where nothing is configured, list as available to deploy
- where something is configured, list only if the configuration meets the request of the user regarding watch config